src/getLocationForJsonPath.ts

```diff
@@ -66,16 +70,22 @@ function getEndPosition(node: YAMLNode): number {
   return node.endPosition;
 }
 
-function findNodeAtPath(node: YAMLNode, path: JsonPath, closest: boolean) {
+function findNodeAtPath(
+  node: YAMLNode,
+  path: JsonPath,
+  { closest, mergeKeys }: { closest: boolean; mergeKeys: boolean }
+) {
   pathLoop: for (const segment of path) {
     switch (node && node.kind) {
       case Kind.MAP:
-        for (const item of node.mappings) {
+        const mappings = getMappings(node.mappings, mergeKeys);
+        for (let i = mappings.length - 1; i >= 0; i--) {
+          const item = mappings[i];
           if (item.key.value === segment) {
             if (item.value === null) {
-              node = item.key;
+              node = item.key as YAMLNode;
             } else {
-              node = item.value;
+              node = item.value as YAMLNode;
```

Comment:
Both the key and the value are Yaml nodes?

Reply:
Same issue as https://github.com/stoplightio/yaml/pull/20#discussion_r309163362